### DISCUSSION TITLE
Odd volume and energy fluctuations in MD simulations with NNP

### DISCUSSION THREAD

**Discussion by ghost** [deepmodeling/deepmd-kit#2418] [General]
Hello everyone,

I recently trained the NNP for molten LiCl and ran MD simulations (DeepmdKit v.2.0.3). The errors in energies and forces are in the order of 10^-3 eV/atom and 10^-2 eV/atom, respectively. However, I found that the values for the volumes (below picture) and energies fluctuate very smoothly. I compared this behavior with previously reported NNPs for LiCl (Journal of Materials Science & Technology 75 (2021) 78–85) from the previously available online deepmd library, but the results do not vary much. Is this normal, as i believe the fluctuations should be less smooth and noisier?

![image](https://user-images.githubusercontent.com/105665149/226899353-22231399-9574-42ae-85d3-734df64129e4.png)

What I've tried:
1) Increasing box size
2) Training where virial stresses are included

**input.json:**

{
"_comment": " model parameters",
"model": {
"type_map": ["Li", "Cl"],
"descriptor" :{
"type": "se_e2_a",
"sel": [100, 100],
"rcut_smth": 6.80,
"rcut": 7.00,
"neuron": [25, 50, 100],
"resnet_dt": false,
"axis_neuron": 16,
"seed": 2,
"_comment": " that's all"
},
"fitting_net" : {
"neuron": [240, 240, 240],
"resnet_dt": true,
"seed": 20,
"_comment": " that's all"
},
"_comment": " that's all"
},
"learning_rate" :{
"type": "exp",
"decay_steps": 5000,
"start_lr": 0.001,
"stop_lr": 3.51e-8,
"_comment": "that's all"
},
"loss" :{
"type": "ener",
"start_pref_e": 0.02,
"limit_pref_e": 1,
"start_pref_f": 1000,
"limit_pref_f": 1,
"start_pref_v": 0,
"limit_pref_v": 0,
"_comment": " that's all"
},
"training" : {
"training_data": {
"systems": ["./training"],
"batch_size": "auto",
"_comment": "that's all"
},
"validation_data": {
"systems": ["./validation"],
"batch_size": "auto",
"numb_btch": 1,
"_comment": "that's all"
},
"numb_steps": 1000000,
"seed": 20,
"disp_file": "lcurve.out",
"disp_freq": 100,
"save_freq": 100,
"_comment": "that's all"
},
"_comment": "that's all"
}

**LAMMPS input:** 

units           metal
boundary        p p p
atom_style      atomic
variable        T equal 1033

neighbor            2.0 bin
neigh_modify    every 1 check yes

read_data          data.licl
mass                1 6.941
mass                2 35.453

pair_style          deepmd graph.pb
pair_coeff          * *

thermo          100
thermo_style    custom step temp pe ke etotal press vol lx density
dump            1 all custom 100 licl.traju id type xu yu zu
dump_modify     1 sort id

velocity        all create $T 2803 mom yes rot yes dist gaussian

fix             1 all npt temp $T $T 100 iso 1.0 1.0 1000
run             8000000
unfix           1

write_data      licl.dat
write_restart   licl.rst

**Comment by owen-rett** [comment]
I suspect the issue is due to the pressure dampening parameter, ```fix 1 all npt temp $T $T 100 iso 1.0 1.0 **1000**``` used in your lammps script. You're using metal units, meaning that the default timestep is 0.001 ps, or 1 fs, so your system is relaxing the pressure over 10^6 timesteps, rather than the typical 1000 timesteps. I've checked the paper, and I can't find a mention of the pressure dampening parameter used, however I'm checking the preprint, and could have missed something, but I'd assume 100 timesteps for the thermal dampening parameter, and 1000 timesteps for the pressure dampening parameter. I've modified the lammps script to reflect this. If this doesn't fix the issue, I'd recommend checking the DP parameters and training set against those used in the paper above. 

```
units metal
boundary p p p
atom_style atomic
variable T equal 1033
variable dt  equal  0.001

neighbor 2.0 bin
neigh_modify every 1 check yes

read_data data.licl
mass 1 6.941
mass 2 35.453

pair_style deepmd graph.pb
pair_coeff * *

thermo 100
thermo_style custom step temp pe ke etotal press vol lx density
dump 1 all custom 100 licl.traju id type xu yu zu
dump_modify 1 sort id

velocity all create $T 2803 mom yes rot yes dist gaussian

fix 1 all npt temp $T $T $(100*dt) iso 1.0 1.0 $(1000*dt)
run 8000000
unfix 1

write_data licl.dat
write_restart licl.rst```

**Comment by ghost** [comment]
Thank you, that resolved the issue with me. I incorrectly assumed that the damping parameter was in timesteps, rather than the time.
